internal/app/api/handler/login/login.go

```diff
@@ -0,0 +1,56 @@
+package login
+
+import (
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/TheFlies/ofriends/internal/app/types"
+	"github.com/TheFlies/ofriends/internal/pkg/glog"
+	"github.com/TheFlies/ofriends/internal/pkg/respond"
+)
+
+type (
+	service interface {
+		GetByName(username string) (*types.Usercache, error)
+		AddUser(user *types.Usercache) error
+		CheckExistence(username string) bool
+		LDAPAuthenticate(username string, password string) (interface{}, error)
```

Comment:
The method should be Authenticate(username string, password string) (interface{}, error)
No specific authentication mechanism should be mentioned here in the handler